hikey.mk

```diff
@@ -179,123 +206,140 @@ optee-linuxdriver-clean: optee-linuxdriver-clean-common
-
-update_rootfs: busybox optee-client optee-linuxdriver filelist-tee linux-gen_init_cpio
+# Read stdin, expand ${VAR} environment variables, output to stdout
+# http://superuser.com/a/302847
+define expand-env-var
+awk '{while(match($$0,"[$$]{[^}]*}")) {var=substr($$0,RSTART+2,RLENGTH -3);gsub("[$$]{"var"}",ENVIRON[var])}}1'
+endef
+
+filelist-all: busybox
 	cat $(GEN_ROOTFS_PATH)/filelist-final.txt | sed '/fbtest/d' > $(GEN_ROOTFS_PATH)/filelist-all.txt
-	cat $(GEN_ROOTFS_PATH)/filelist-all.txt $(GEN_ROOTFS_PATH)/filelist-tee.txt > $(GEN_ROOTFS_PATH)/filelist.tmp
+	export KERNEL_VERSION=$(call KERNEL_VERSION); \
+	export TOP=$(ROOT); export MULTIARCH=$(MULTIARCH); \
+	$(expand-env-var) <$(PATCHES_PATH)/rootfs/initramfs-add-files.txt >> $(GEN_ROOTFS_PATH)/filelist-all.txt; \
+	find $(OPTEE_TEST_OUT_PATH) -type f -name "xtest" | sed 's/\(.*\)/file \/bin\/xtest \1 755 0 0/g' >> $(GEN_ROOTFS_PATH)/filelist-all.txt; \
+	find $(OPTEE_TEST_OUT_PATH) -name "*.ta" | \
+		sed 's/\(.*\)\/\(.*\)/file \/lib\/optee_armtz\/\2 \1\/\2 444 0 0/g' >> $(GEN_ROOTFS_PATH)/filelist-all.txt
+
+update_rootfs: optee-client optee-linuxdriver xtest aes-perf sha-perf strace filelist-all linux-gen_init_cpio
 	cd $(GEN_ROOTFS_PATH); \
-	        $(LINUX_PATH)/usr/gen_init_cpio $(GEN_ROOTFS_PATH)/filelist.tmp | gzip > $(GEN_ROOTFS_PATH)/filesystem.cpio.gz
+	        $(LINUX_PATH)/usr/gen_init_cpio $(GEN_ROOTFS_PATH)/filelist-all.txt | gzip > $(GEN_ROOTFS_PATH)/filesystem.cpio.gz
 
-.PHONY: update_rootfs_clean
-update_rootfs_clean:
-	cd $(GEN_ROOTFS_PATH); \
-	rm -f $(GEN_ROOTFS_PATH)/filesystem.cpio.gz $(GEN_ROOTFS_PATH)/filelist.tmp $(GEN_ROOTFS_PATH)/filelist-tee.txt $(GEN_ROOTFS_PATH)/filelist-all.txt; \
-	if [ -f "$(USBNETSH_PATH)" ]; then rm $(USBNETSH_PATH); fi;
+.PHONY: update_rootfs-clean
+update_rootfs-clean:
+	rm -f $(GEN_ROOTFS_PATH)/filesystem.cpio.gz $(GEN_ROOTFS_PATH)/filelist-all.txt $(GEN_ROOTFS_PATH)/filelist-tmp.txt
+
+################################################################################
+# grub
+################################################################################
+grub-flags := CC="$(CCACHE)gcc" \
+	TARGET_CC="$(AARCH64_CROSS_COMPILE)gcc" \
+	TARGET_OBJCOPY="$(AARCH64_CROSS_COMPILE)objcopy" \
+	TARGET_NM="$(AARCH64_CROSS_COMPILE)nm" \
+	TARGET_RANLIB="$(AARCH64_CROSS_COMPILE)ranlib" \
+	TARGET_STRIP="$(AARCH64_CROSS_COMPILE)strip"
+
+.PHONY: grub
+grub: prepare
+	cd $(GRUB_PATH); \
+	./autogen.sh; \
+	./configure --target=aarch64 --enable-boot-time $(grub-flags); \
+	$(MAKE) -C $(GRUB_PATH); \
+	./grub-mkimage \
+		--verbose \
+		--output=$(OUT_PATH)/grubaa64.efi \
+		--config=$(PATCHES_PATH)/grub/grub.configfile \
+		--format=arm64-efi \
+		--directory=grub-core \
+		--prefix=/boot/grub \
+		boot chain configfile efinet ext2 fat gettext help linux loadenv lsefi normal part_gpt part_msdos read search search_fs_file search_fs_uuid search_label terminal terminfo tftp time
```

Comment:
Got the list from B&B team's build script, but yes, figured it won't hurt. There were actually only 6 modules too originally, but the updated grub.configfile is using the search* modules, so it seems better to just include all of them, in case more modules need to be used later, then no changes to the build are needed.